repose-aggregator/components/filters/merge-header/src/test/scala/org/openrepose/filters/mergeheader/MergeHeaderFilterTest.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.mergeheader
+
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import org.junit.runner.RunWith
+import org.mockito.Matchers.{any, anyString}
+import org.mockito.Mockito.verify
+import org.openrepose.core.services.config.ConfigurationService
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+import org.springframework.mock.web.{MockFilterChain, MockFilterConfig, MockHttpServletRequest, MockHttpServletResponse}
+
+import scala.collection.JavaConversions._
+
+@RunWith(classOf[JUnitRunner])
+class MergeHeaderFilterTest extends FunSpec with BeforeAndAfter with Matchers with MockitoSugar {
+
+  var configurationService: ConfigurationService = _
+  var filter: MergeHeaderFilter = _
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: MockHttpServletResponse = _
+  var filterChain: MockFilterChain = _
+
+  before {
+    configurationService = mock[ConfigurationService]
+
+    servletRequest = new MockHttpServletRequest
+    servletResponse = new MockHttpServletResponse
+    filterChain = new MockFilterChain
+
+    filter = new MergeHeaderFilter(configurationService)
+  }
+
+  describe("init") {
+    it("should register a configuration listener") {
+      filter.init(new MockFilterConfig("MergeHeaderFilter"))
+
+      verify(configurationService).subscribeTo(anyString(), anyString(), any(), any(), any[Class[MergeHeaderConfig]])
+    }
+  }
+
+  describe("destroy") {
+    it("should unregister a configuration listener") {
+      filter.destroy()
+
+      verify(configurationService).unsubscribeFrom(anyString(), any())
+    }
+  }
+
+  describe("doFilter") {
```

Comment:
I was really hoping to see some tests for when the configured header isn't in the request/response.